hoomd/Variant.cc

```diff
@@ -3,12 +3,24 @@
 
 #include "Variant.h"
 
-//* Method to enable unit testing of C++ trigger calls from pytest
+/// Method to enable unit testing of C++ variant calls from pytest
 Scalar testVariantCall(std::shared_ptr<Variant> t, uint64_t step)
     {
     return (*t)(step);
     }
 
+/// Method to enable unit testing of C++ variant min class from pytest
+Scalar testVariantMin(std::shared_ptr<Variant> t)
+    {
+    return t->min();
+    }
+
+/// Method to enable unit testing of C++ variant max class from pytest
+Scalar testVariantMax(std::shared_ptr<Variant> t)
+    {
+    return t->max();
+    }
```

Comment:
I would rename these to `get*` since they're not testing methods, they're just getters that you expose. But also, why can't you just call the object's min and max methods? pybind should understand the inheritance part just fine.

Reply:
I assume you've tried this and it doesn't work, just trying to understand why.

Reply:
This is purely for testing that the min and max calls work in C++ explicitly for custom variants. Otherwise, we would not be able to directly test that the method calls work in C++.

Reply:
Don't custom variants also expose their min and max methods to Python though? Also, still suggesting the rename.

Reply:
They expose them as `_min` and `_max` which allow us to use properties for `min` and `max` on the Python side even for custom variants. Also, they are named `testVariant{Method}` because the sole purpose is to test that custom `Variant` classes' `min` and `max` methods work correctly. We could name them `get{Method}`, but I feel that is less clear to future developers.

Reply:
I added some developer comments to clear up the purpose of the methods.

Reply:
OK, I'm clearer on the goals here now and am on board with this.